### ISSUE TITLE
Fix unattended-upgrades conflict

### ISSUE THREAD

**Issue by conorsch** [freedomofpress/securedrop#1032]
The install docs contain instructions to disable `unattended-upgrades` during initial setup, since we use `cron-apt` to handle automatic updates (see #717). However, the Ansible playbooks do not enforce absence of `unattended-upgrades`, which can cause breakage if the admin leaves it enabled. Specifically, `unattended-upgrades` does not by default choose to preserve manually edited configuration files, and instead prompts for admin intervention. During an automatic update, prompting  will result in failed updates and a broken apt state, blocking future package updates until the admin manually reconfigures the broken packages.

For initial remediation, the playbooks should be updated to remove `unattended-upgrades` during configuration of `cron-apt`. At a later date (0.4?) the choice of `cron-apt` over `unattended-upgrades` can be reconsidered.

**Comment by ageis** [comment]
I agree the choice can be reconsidered, I think in the switch to cron-apt there may have been a misunderstanding in how to pull updates from multiple origins with unattended-upgrades which is perfectly sufficient (and OS default), it doesn't require regex.

**Comment by conorsch** [comment]
@ageis I'm a fan of reconsidering, but right now I'm not sure this issue is adequately diagnosed. 

As a concrete example of a dpkg prompt, a recent update to `libpam-systemd` causes a prompt for preserving changes to `/etc/pam.d/common-auth`, due to the Google 2FA changes made by the `restrict_direct_access_*` roles. The prompt is visible when running an upgrade interactively via the command line:  `apt-get install --only-upgrade pam-systemd` 

(To downgrade to the previous version of the package for testing, run `apt-get install libpam-systemd=204-5ubuntu20`, or provision new staging boxes with `ansible.skip_tags =  [ "install_local_pkgs", "apt-upgrade" ]` set in the Vagrantfile.)

**However,** the dpkg prompt only appears when the upgrade is run interactively from the command-line. Neither `cron-apt` nor `unattended-upgrades` will cause this package to be updated, since the patch comes from `trusty-updates/main` rather than `trusty-security/main`.

Inspecting `/usr/bin/unattended-upgrade` shows that it responsibly tries to avoid upgrading any packages that require dpkg prompts, unless the unattended-upgrade config explicitly declares `DPkg::Options` values to override this cautious behavior. Therefore I'm not convinced that unattended-upgrades is going to break this. 

I've set up a virtual test instance with intentionally outdated packages so I can try to catch this behavior in the apt logs.

**Comment by conorsch** [comment]
Logs from test instance:

> 2015-05-28 23:14:34,082 INFO Initial blacklisted packages: 
2015-05-28 23:14:34,083 INFO Starting unattended upgrades script
2015-05-28 23:14:34,084 INFO Allowed origins are: ['o=Ubuntu,a=trusty-security']
2015-05-28 23:14:35,459 INFO No packages found that can be upgraded unattended

A manual upgrade still lists the `libpam-systemd` package as a pending update:

```
root@app-staging:~# apt-get -s upgrade | grep libpam-systemd
  isc-dhcp-common libaccountsservice0 libpam-systemd libsystemd-daemon0
Inst libpam-systemd [204-5ubuntu20.11] (204-5ubuntu20.12 Ubuntu:14.04/trusty-updates [amd64]) []
Conf libpam-systemd (204-5ubuntu20.12 Ubuntu:14.04/trusty-updates [amd64])
```

There's some other package conflict somewhere that may cause a broken apt state. I'll start stepping through likely candidates in the security list and try to recreate a conflict.

**Comment by conorsch** [comment]
After reviewing logs, it looks like any recent apt breakage was caused by the /boot partition filling up on long-running hosts. A task was added to remove default kernels in 3fdffec and is already included in the latest releases (since 0.3.1).

Any prompt for conf files related to `libpam-systemd` only appears when the admin runs an upgrade interactively. After quite a bit of testing, I wasn't able to reproduce a package state where `unattended-upgrades` caused breaking changes—which is a relief. The `unattended-upgrades` package appears to play nicely with `cron-apt`, at least when `cron-apt` is configured with the settings our Ansible playbooks enforce.

I don't see any stability issues for production instances, so I'm closing this issue. (@dolanjs holler if you have data to the contrary.)

**Comment by garrettr** [comment]
@conorsch Thanks for continuing to investigate that. Everything makes so much sense now.
